apk-tools/usrmerge-lib.patch

```
@@ -0,0 +1,33 @@
+commit 65c066c3db62b75cf5659e216aa4fb838db58ec9 (HEAD)
+Author: Scott Moser <smoser@brickies.net>
+Date:   Thu Mar 27 16:53:48 2025 -0400
+
+    try initdb in usrmerge-lib world
+
+diff --git a/src/database.c b/src/database.c
+index dc5e4fd..a37123c 100644
+--- a/src/database.c
++++ b/src/database.c
+@@ -1288,7 +1288,7 @@ static int add_protected_paths_from_file(void *ctx, int dirfd, const char *file)
+ 
+ static int apk_db_create(struct apk_database *db)
+ {
+-	int fd;
++	int fd, r;
+ 
+ 	mkdirat(db->root_fd, "tmp", 01777);
+ 	mkdirat(db->root_fd, "dev", 0755);
+@@ -1299,6 +1299,13 @@ static int apk_db_create(struct apk_database *db)
+ 	mknodat(db->root_fd, "dev/console", S_IFCHR | 0600, makedev(5, 1));
+ 	mkdirat(db->root_fd, "etc", 0755);
+ 	mkdirat(db->root_fd, "etc/apk", 0755);
++	r = symlinkat("usr/lib", db->root_fd, "lib");
++	if (r == -1) {
++	apk_error("failed to symlink lib -> usr/lib");
++		return -errno;
++	}
++	mkdirat(db->root_fd, "usr", 0755);
++	mkdirat(db->root_fd, "usr/lib", 0755);
```

Just a thought...I had. I'm not sure what is right.
They basically just ignore all errors here.  I think the intent is to not fail on subsequent apk runs that have '--initdb' in attempt to be more "user-friendly".  Ultimately here they just take the path that if they can open etc/apk/world and lib/apk/db/installed then they're happy.

I chose to error above if the symlink fails.  I think that isn't right.

As it is here
1. subsequent calls to '--initdb' will fail because the symlinkat will return EEXIST.
2. if lib exists as a directory it will fail.

I'm pretty sure we to ignore the error in 1.
I'm not sure what we want to do in 2.

Do we want '--initdb' to work on a non-usrmerged lib?  thoughts?

---

I think we should make case 1 succeed, I am indifferent on making case 2 succeed but if ignoring the return value will make both work I think that's fine.